ext/ldap/ldap.c

```diff
@@ -987,6 +987,17 @@ PHP_FUNCTION(ldap_connect)
 			snprintf( url, urllen, "ldap://%s:" ZEND_LONG_FMT, host, port );
 		}
 
+#ifdef LDAP_OPT_X_TLS_NEWCTX
+		if (!memcmp(url, "ldaps:", 6)) {
```

Comment:
Note that `url` can be NULL, because `url = host` and `host` is nullable.
Also I prefer if you use a normal string comparison rather than `memcmp` to avoid potential overreads. I don't know if that's possible (depends on the behaviour of `ldap_is_ldap_url`) but better safe than sorry and it wouldn't be really that much of a performance difference anyway.

Reply:
done